Assets/_Project/Scripts/UserInterface/Components/AppearanceChangingButton.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using UnityEngine.Serialization;
+using UnityEngine.UI;
+
+namespace PolSl.UrbanHealthPath.UserInterface.Components
+{
+    [RequireComponent(typeof(Button))]
+    public class AppearanceChangingButton : MonoBehaviour
+    {
+        [Serializable] public struct ButtonAppearance
+        {
+            [SerializeField] private Sprite sprite;
+            [SerializeField] private String text;
+            [SerializeField] private int bottomOffset;
+
+            public Sprite GetSprite()
+            {
+                return sprite;
+            }
+
+            public String GetText()
+            {
+                return text;
+            }
+            public int GetOffset()
+            {
+                return bottomOffset;
+            }
+        }
+
+        [Tooltip("List with button appearances, that changes every time user click on the button. " +
+                 "First list element is a default button appearance. ")]
+        [FormerlySerializedAs("_buttonAppearances")] 
+        [SerializeField] private List<ButtonAppearance> buttonAppearances;
```

Comment:
Shouldn't a private field start with an underscore?

Reply:
It was named like that but then renamed? What happened?

Reply:
underscores added with FormerlySerializedAs tag.